### PR TITLE
support Llama2-chat fine-tuning

### PR DESCRIPTION
**Pull request by csuhan**

### CODE REVIEW

**Comment by tmm1** [comment]
Thanks for getting this started! I am interested in llama2-chat fine-tuning to take advantage of Ghost Attention (GAtt).

One quirk of the llama2 chat format is the inclusion of a EOS/BOS token pair in between conversation turns. I couldn't tell if this was currently being inserted in this PR.

For example see:

https://github.com/lm-sys/FastChat/pull/1995/files
https://github.com/lm-sys/FastChat/pull/1997/files

and more info on: https://gpus.llm-utils.org/llama-2-prompt-template/

File: accessory/data/llama2_chat.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+B_INST, E_INST = "[INST]", "[/INST]"
+B_SYS, E_SYS = "<<SYS>>\n", "\n<</SYS>>\n\n"
+DEFAULT_SYSTEM_PROMPT = """\
+You are a helpful, respectful and honest assistant. Always answer as helpfully as possible, while being safe. Your answers should not include any harmful, unethical, racist, sexist, toxic, dangerous, or illegal content. Please ensure that your responses are socially unbiased and positive in nature.
+
+If a question does not make any sense, or is not factually coherent, explain why instead of answering something not correct. If you don't know the answer to a question, please don't share false information."""
+
+
+class FinetuneLlam2ChatDataset(Dataset):
```
Comment (tmm1) [inline accessory/data/llama2_chat.py:23]:
```suggestion
class FinetuneLlama2ChatDataset(Dataset):
```

File: accessory/main_finetune.py [hunk 2]
```diff
@@ -37,6 +37,7 @@
 from torch.utils.data import Dataset
 from data.alpaca import FinetuneDataset, transform_train, FinetuneDistSampler
 from data.conversation.dataset import FinetuneDialogDataset
+from data.llama2_chat import FinetuneLlam2ChatDataset
```
Comment (tmm1) [inline accessory/main_finetune.py:40]:
```suggestion
from data.llama2_chat import FinetuneLlama2ChatDataset
```

File: accessory/main_finetune.py [hunk 3]
```diff
@@ -219,7 +222,12 @@ def main(args):
 
     # data
     if args.dialog:
-        DatasetClass = FinetuneDialogDataset
+        if not args.with_llama2_chat:
+            # use Vicuna's format
+            DatasetClass = FinetuneDialogDataset
+        else:
+            # use llama2_chat's format
+            DatasetClass = FinetuneLlam2ChatDataset
```
Comment (tmm1) [inline accessory/main_finetune.py:230]:
```suggestion
            DatasetClass = FinetuneLlama2ChatDataset
```

**Review by tmm1 — COMMENTED** [review]
I fixed a small typo
